src/features/class-rest-api.php

```diff
@@ -23,26 +23,127 @@ public function __construct() {}
 	 * Boot the feature.
 	 */
 	public function boot(): void {
-		add_filter( 'rest_post_query', [ $this, 'add_type_param' ], 10, 2 );
+		add_action( 'rest_api_init', [ $this, 'register_endpoints' ] );
 	}
 
 	/**
-	 * Add post_type to rest post query if the type param is set.
+	 * Sets up the endpoint.
 	 *
-	 * @param array<array<int, string>|string> $query_args The existing query args.
-	 * @param WP_REST_Request                  $request The REST request.
-	 * @return array<array<int, string>|string>
+	 * @return void
 	 */
-	// @phpstan-ignore-next-line
-	public function add_type_param( $query_args, $request ): array { // phpcs:ignore Squiz.Commenting.FunctionComment.WrongStyle
-		$type = $request->get_param( 'type' );
-
-		if ( ! empty( $type ) && is_string( $type ) ) {
-			$types                   = explode( ',', $type );
-			$types                   = array_filter( $types, 'post_type_exists' );
-			$query_args['post_type'] = $types;
+	public function register_endpoints(): void {
+		register_rest_route(
+			'wp-curate/v1',
+			'/posts/',
+			[
+				'methods'             => 'GET',
+				'callback'            => [ $this, 'get_posts' ],
+				'permission_callback' => function () {
+					return current_user_can( 'edit_posts' );
+				},
+			]
+		);
+	}
+
+	/**
+	 * Gets the posts.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 * @return array<int> The post IDs.
+	 */
+	public function get_posts( WP_REST_Request $request ): array { // phpcs:ignore Squiz.Functions.MultiLineFunctionDeclaration.ContentAfterBrace @phpstan-ignore-line
+		$search_term      = $request->get_param( 'search' ) ?? '';
+		$offset           = $request->get_param( 'offset' ) ?? 0;
+		$post_type_string = $request->get_param( 'post_type' ) ?? 'post';
+		$per_page         = $request->get_param( 'per_page' ) ?? 20;
+		$trending         = 'true' === $request->get_param( 'trending' ) ? true : false;
+		$tax_relation     = $request->get_param( 'tax_relation' ) ?? 'OR';
+
+		if ( ! is_string( $post_type_string ) ) {
+			$post_type_string = 'post';
+		}
+
+		/**
+		 * Filters the allowed taxonomies.
+		 *
+		 * @param array<string> $allowed_taxonomies The allowed taxonomies.
+		 * @return array<string> The allowed taxonomies.
```

Comment:
```suggestion
```

This could alternatively use the core standard for duplicate hook docs: https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-hooks-actions-and-filters